ride-share.rb

```diff
@@ -0,0 +1,101 @@
+rides =
+  {"DR0001" => [{:date => "3rd Feb 2016", :cost => 10, :rider_id => "RD0003", :rating => 3},
+                {:date => "3rd Feb 2016", :cost => 30, :rider_id => "RD0015", :rating => 4},
+                {:date => "5th Feb 2016", :cost => 45, :rider_id => "RD0003", :rating => 2}],
+   "DR0002" => [{:date => "3rd Feb 2016", :cost => 25, :rider_id => "RD0073", :rating => 5},
+                {:date => "4th Feb 2016", :cost => 15, :rider_id => "RD0013", :rating => 1},
+                {:date => "5th Feb 2016", :cost => 35, :rider_id => "RD0066", :rating => 3}],
+   "DR0003" => [{:date => "4th Feb 2016", :cost => 5, :rider_id => "RD0066", :rating => 5},
+                {:date => "5th Feb 2016", :cost => 50, :rider_id => "RD0003", :rating => 2}],
+   "DR0004" => [{:date => "3rd Feb 2016", :cost => 5, :rider_id => "RD0022", :rating => 5},
+                {:date => "4th Feb 2016", :cost => 10, :rider_id => "RD0022", :rating => 4},
+                {:date => "5th Feb 2016", :cost => 20, :rider_id => "RD0073", :rating => 5}]}
+
+puts "The Number of Rides Each Driver Has Given:\n"
+number_of_rides = {}
+rides.each do |drivers, rides|
+  number_of_rides[drivers] = rides.length.to_f
+end
+number_of_rides.each do |drivers, rides|
+  puts "#{drivers} has driven #{rides.to_i} rides"
+end
+
+puts "\nThe Total Amount of Money Each Driver Has Made:\n"
+pay_hash = {}
+rides.each do |drivers, rides|
+  money_made = 0
+  rides.each do |eachride|
+    money_made += eachride[:cost]
+  end
+  pay_hash[drivers] = money_made
+end
+
+pay_hash.each do |drivers, pay|
+  puts "The total amount of money Driver #{drivers} made was $#{pay}"
+end
+
+puts "\nThe Average Rating for Each Driver:\n"
+rating_hash = {}
+rides.each do |drivers, rides|
```

Comment:
this code is well done, but is there any way you could leverage methods to improve clarity?